macros/extract_descriptor.sql

```diff
@@ -1,4 +1,42 @@
 -- grab descriptor codes from namespaced descriptor values
 {% macro extract_descriptor(col) -%}
-    split_part({{ col }}, '#', -1)
+    
+    {%- set stripped_col = col.split(":")[-3] -%}
+    {%- set config = var('descriptors')[stripped_col] or None -%}
```

Comment:
This still raises an error when the config is not defined. I'd recommend something of the following:
```
{%- set config = var('descriptors', {}).get(stripped_col) or None -%}
```

Reply:
good catch, resolved